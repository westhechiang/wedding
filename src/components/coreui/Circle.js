import styled from 'styled-components';
import { theme } from '../../ui';

const Circle = styled.div`
  height: ${props => props.size};
  width: ${props => props.size};
  border-radius: 50%;
  border: ${props => (props.borderColor ? theme.border(props.borderColor) : 'none')};
  overflow: hidden;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.backgroundColor};
  color: ${props => props.color};
  text-align: ${props => props.textAlign};
  ${props => (props.hover ? '&:hover { cursor: pointer; }' : '')}
  ${props => (props.active ? `&:active { background-color: ${theme.color.grayDark}; }` : '')}
`;

export default Circle;
