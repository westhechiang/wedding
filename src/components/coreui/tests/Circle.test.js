import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Circle from '../../../components/coreui/Circle';

describe('<Circle />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<Circle />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
