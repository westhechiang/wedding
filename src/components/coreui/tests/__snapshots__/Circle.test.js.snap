// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa OprAx"
/>
`;
